app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivity.kt

```diff
@@ -61,34 +68,80 @@ class AddPictureActivity : AppCompatActivity() {
         }
     }
 
-    private fun cameraIsAvailable(): Boolean {
-        return applicationContext.packageManager.hasSystemFeature(PackageManager.FEATURE_CAMERA_ANY)
-    }
+    /**
+    The ActivityResultContract which should be used when launching this activity.
+    The launch argument is an Array<String> containing the categories to select from.
+    The return value is a Pair containing the selected category as a first element and the Uri pointing to the image as a second element
+     */
+    object AddPictureContract : ActivityResultContract<Array<String>, Pair<String, Uri>?>() {
```

Comment:
To represent the categories to select from, wouldn't it be better to use the Category class ? or am I missing something?
```suggestion
    object AddPictureContract : ActivityResultContract<Set<Category>, Pair<String, Uri>?>() {
```

Reply:
Yes that would be better indeed. I wanted Category to be finalized before adapting it.